### PR TITLE
Add stageEntityChange() method to display difference when creating a reserved list 

### PR DESCRIPTION
**Pull request by rachelguan**
similar to adding stageEntityChange() method to UpdateReservedList.java.

---
This change is [Reviewable](https://reviewable.io/reviews/google/nomulus/1149)

### CODE REVIEW

**Review by mindhog — CHANGES_REQUESTED** [review]
> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1149)* status: all files reviewed, 2 unresolved discussions (waiting on @rachelguan, @sarahcaseybot, and @weiminyu)

___
*[core/src/main/java/google/registry/tools/CreateReservedListCommand.java, line 73 at r1](https://reviewable.io/reviews/google/nomulus/1149#-M_VtwIF7uV_MNVMYX4B:-M_VtwIG-CoYqnL6wBSN:bh4gd33) ([raw file](https://github.com/google/nomulus/blob/ae1076ff0d17284be23404cf37613441e11ff7e5/core/src/main/java/google/registry/tools/CreateReservedListCommand.java#L73)):*
> ```Java
>     // ReservedList is a sqlEntity; it triggers the static method Vkey.create(Key<?> ofyCall),
>     // which invokes a static ReservedList.createVkey(Key ofyKey) method that does not exist.
>     // the sql primary key field (revisionId) is only set when it's being persisted;
> ```

I'm confused by this comment.  It looks like you're explaining why you're explicitly creating a VKey, but it seems to go on a long journey to get there.  For one thing, I think you mean `VKey.from()` rather than `VKey.create()` ("from" is the one that does the magical dispatching to a static method).

I would shorten this to something like "we have to create an ofy-only vkey here because the sql primary key field (revisionId) is only set when the object is persisted."
___
*[core/src/test/java/google/registry/tools/CreateReservedListCommandTest.java, line 209 at r1](https://reviewable.io/reviews/google/nomulus/1149#-M_VyZ1W5yFYCDQQhJ4s:-M_VyZ1W5yFYCDQQhJ4t:b-g7g6c7) ([raw file](https://github.com/google/nomulus/blob/ae1076ff0d17284be23404cf37613441e11ff7e5/core/src/test/java/google/registry/tools/CreateReservedListCommandTest.java#L209)):*
> ```Java
>     File tmpFile = tmpDir.resolve("xn--q9jyb4c_common-tmp.txt").toFile();
>     String fileStr = "";
>     Files.asCharSink(tmpFile, UTF_8).write(fileStr);
> ```

I think it would be a lot easier to write this purely in terms of Paths:
```
Path tmpFile = tmpDir.resolve("...");
Files.createFile(tmpFile);
...
command.input = tmpFile;
```

**Review by rachelguan — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1149)* status: 0 of 2 files reviewed, 2 unresolved discussions (waiting on @mindhog, @sarahcaseybot, and @weiminyu)

___
*[core/src/main/java/google/registry/tools/CreateReservedListCommand.java, line 73 at r1](https://reviewable.io/reviews/google/nomulus/1149#-M_VtwIF7uV_MNVMYX4B:-M_WGrUv3Hj6j1cX8swa:bapv2lx) ([raw file](https://github.com/google/nomulus/blob/ae1076ff0d17284be23404cf37613441e11ff7e5/core/src/main/java/google/registry/tools/CreateReservedListCommand.java#L73)):*
<details><summary><i>Previously, mindhog (Michael Muller) wrote…</i></summary><blockquote>

I'm confused by this comment.  It looks like you're explaining why you're explicitly creating a VKey, but it seems to go on a long journey to get there.  For one thing, I think you mean `VKey.from()` rather than `VKey.create()` ("from" is the one that does the magical dispatching to a static method).

I would shorten this to something like "we have to create an ofy-only vkey here because the sql primary key field (revisionId) is only set when the object is persisted."
</blockquote></details>

i see. i just used the comment i wrote previous for update and followed same structure as update. I used Vkey.create() previously. Vkey.from(Key<T> key) takes a key, but key is not created, so hows that going to work? 
___
*[core/src/test/java/google/registry/tools/CreateReservedListCommandTest.java, line 209 at r1](https://reviewable.io/reviews/google/nomulus/1149#-M_VyZ1W5yFYCDQQhJ4s:-M_WlVZ10AVhjVZCjLPP:b-896fix) ([raw file](https://github.com/google/nomulus/blob/ae1076ff0d17284be23404cf37613441e11ff7e5/core/src/test/java/google/registry/tools/CreateReservedListCommandTest.java#L209)):*
<details><summary><i>Previously, mindhog (Michael Muller) wrote…</i></summary><blockquote>

I think it would be a lot easier to write this purely in terms of Paths:
```
Path tmpFile = tmpDir.resolve("...");
Files.createFile(tmpFile);
...
command.input = tmpFile;
```
</blockquote></details>

Done.

**Review by weiminyu — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1149)* status: 0 of 2 files reviewed, 2 unresolved discussions (waiting on @mindhog and @sarahcaseybot)

___
*[core/src/main/java/google/registry/tools/CreateReservedListCommand.java, line 73 at r1](https://reviewable.io/reviews/google/nomulus/1149#-M_VtwIF7uV_MNVMYX4B:-M_WwvOY1-VZqYytSwCs:bw2p0dx) ([raw file](https://github.com/google/nomulus/blob/ae1076ff0d17284be23404cf37613441e11ff7e5/core/src/main/java/google/registry/tools/CreateReservedListCommand.java#L73)):*
<details><summary><i>Previously, rachelguan (Rachel Guan) wrote…</i></summary><blockquote>

i see. i just used the comment i wrote previous for update and followed same structure as update. I used Vkey.create() previously. Vkey.from(Key<T> key) takes a key, but key is not created, so hows that going to work? 
</blockquote></details>

VKey.from() takes an ofy key and expects the 'Kind' class to have a createVKey() method defined. This is not the case with ReservedList, and it is not obvious how to define a createVKey method there when the sql primary key may or may not be available. Using VKey.createOfy() directly seems the only option right now.

**Review by mindhog — CHANGES_REQUESTED** [review]
> Reviewed 1 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1149)* status: 1 of 2 files reviewed, 2 unresolved discussions (waiting on @mindhog, @rachelguan, @sarahcaseybot, and @weiminyu)

___
*[core/src/main/java/google/registry/tools/CreateReservedListCommand.java, line 73 at r1](https://reviewable.io/reviews/google/nomulus/1149#-M_VtwIF7uV_MNVMYX4B:-M_Wyg1SCTbokgWs7GJc:b-dv2wpn) ([raw file](https://github.com/google/nomulus/blob/ae1076ff0d17284be23404cf37613441e11ff7e5/core/src/main/java/google/registry/tools/CreateReservedListCommand.java#L73)):*
<details><summary><i>Previously, weiminyu (Weimin Yu) wrote…</i></summary><blockquote>

VKey.from() takes an ofy key and expects the 'Kind' class to have a createVKey() method defined. This is not the case with ReservedList, and it is not obvious how to define a createVKey method there when the sql primary key may or may not be available. Using VKey.createOfy() directly seems the only option right now.
</blockquote></details>

To be clear: VKey.from() should only require the static createVKey() method for a composite key (an ofy key that has a parent) and that doesn't appear to be the case here.  I don't recall what stageEntityChange() does, so this may or may not matter.

In any case, the updated comment is fine :-)
___
*[core/src/test/java/google/registry/tools/CreateReservedListCommandTest.java, line 209 at r1](https://reviewable.io/reviews/google/nomulus/1149#-M_VyZ1W5yFYCDQQhJ4s:-M_WzdL82s4l_FE6S2Ja:bjnop81) ([raw file](https://github.com/google/nomulus/blob/ae1076ff0d17284be23404cf37613441e11ff7e5/core/src/test/java/google/registry/tools/CreateReservedListCommandTest.java#L209)):*
<details><summary><i>Previously, rachelguan (Rachel Guan) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Sorry, I think you missed part of my suggestion -- I believe you should just be able to do `Files.createFile(tmpFile)` to create an empty file, no?  In the worst case, `Files.write(tmpFile, new byte[0])` would also work.

**Review by mindhog — COMMENTED** [review]
> Reviewed 1 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1149)* status: all files reviewed, 2 unresolved discussions (waiting on @rachelguan, @sarahcaseybot, and @weiminyu)

**Review by rachelguan — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1149)* status: 1 of 2 files reviewed, 2 unresolved discussions (waiting on @mindhog, @sarahcaseybot, and @weiminyu)

___
*[core/src/main/java/google/registry/tools/CreateReservedListCommand.java, line 73 at r1](https://reviewable.io/reviews/google/nomulus/1149#-M_VtwIF7uV_MNVMYX4B:-M_X5ZayCbFRDUCB08pd:bax9up8) ([raw file](https://github.com/google/nomulus/blob/ae1076ff0d17284be23404cf37613441e11ff7e5/core/src/main/java/google/registry/tools/CreateReservedListCommand.java#L73)):*
<details><summary><i>Previously, mindhog (Michael Muller) wrote…</i></summary><blockquote>

To be clear: VKey.from() should only require the static createVKey() method for a composite key (an ofy key that has a parent) and that doesn't appear to be the case here.  I don't recall what stageEntityChange() does, so this may or may not matter.

In any case, the updated comment is fine :-)
</blockquote></details>

stageEntityChange() is used to display the difference between existing list and updated list. existing list/updated list can be null (create/delete). 
___
*[core/src/test/java/google/registry/tools/CreateReservedListCommandTest.java, line 209 at r1](https://reviewable.io/reviews/google/nomulus/1149#-M_VyZ1W5yFYCDQQhJ4s:-M_X6F82C7eEsTpoogRa:bbsetor) ([raw file](https://github.com/google/nomulus/blob/ae1076ff0d17284be23404cf37613441e11ff7e5/core/src/test/java/google/registry/tools/CreateReservedListCommandTest.java#L209)):*
<details><summary><i>Previously, mindhog (Michael Muller) wrote…</i></summary><blockquote>

Sorry, I think you missed part of my suggestion -- I believe you should just be able to do `Files.createFile(tmpFile)` to create an empty file, no?  In the worst case, `Files.write(tmpFile, new byte[0])` would also work. 
</blockquote></details>

Files.createFile(tmpFile) is not there so i went with Files.write(new byte[0], tmpPath.toFile())

**Review by mindhog — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1149)* status: 1 of 2 files reviewed, 1 unresolved discussion (waiting on @mindhog, @sarahcaseybot, and @weiminyu)

___
*[core/src/test/java/google/registry/tools/CreateReservedListCommandTest.java, line 209 at r1](https://reviewable.io/reviews/google/nomulus/1149#-M_VyZ1W5yFYCDQQhJ4s:-M_X7sSzEK_XzcEJHaje:bwppw74) ([raw file](https://github.com/google/nomulus/blob/ae1076ff0d17284be23404cf37613441e11ff7e5/core/src/test/java/google/registry/tools/CreateReservedListCommandTest.java#L209)):*
<details><summary><i>Previously, rachelguan (Rachel Guan) wrote…</i></summary><blockquote>

Files.createFile(tmpFile) is not there so i went with Files.write(new byte[0], tmpPath.toFile())
</blockquote></details>

Ah, looks like I was looking at java.nio.file.Files docs instead of the Google one...
